airflow/migrations/versions/33ae817a1ff4_add_kubernetes_resource_checkpointing.py

```diff
@@ -44,11 +44,11 @@ def upgrade():
     conn = op.get_bind()
 
     # alembic creates an invalid SQL for mssql and mysql dialects
-    if conn.dialect.name in ("mysql"):
+    if conn.dialect.name in {"mysql"}:
```

Comment:
That didn't quite work - there are still changes.

Reply:
Yeah. The conditions were actually wrong so I fixed them :). 

The "in" operator for strings tests for substring match:
```
For the string and bytes types, x in y is True if and only if x is a substring of y. An equivalent test is y.find(x) != -1. Empty strings are always considered to be a substring of any other string, so "" in "abc" will return True.
```

The original condition `conn.dialect.name in ("mysql")` is equivalent to `conn.dialect.name in "mysql"` - the parenthesis are superfluous and they suggest testing `in` against a tuple. I suspect that the original test was actually with the tuple originally ("mysql", "mssql") and someone split the if to mysql/mssql later on but did not notice that the tuple was removed along the way.

Good test in this case would have been `conn.dialect.name in ("mysql",)` (note the coma), but because it is so easy miss the coma and it's not at all obvious, I prefer matching against set rather than tuple (there is no way to make similar mistake in set) - that's why I modified it to use sets in both cases.

It was working fine before - of course - as substrings tested were not ambiguous across different dialects but nevertheless, I think it's worth to fix it (but It would be too much overhead to create a separate issue/commit just to fix those). 

What do you think @feluelle ?

Reply:
Agree with you. I am fine with changing this if it fixes an issue.

But wouldn't that mean that every time we are re-running alembic we need to make this change? (sorry, haven't looked into alembic at all)
What gets actually auto generated?

Reply:
I think it was manually modified after it was generated. Which is quite typical for those kind of scripts/

The migration scripts from Alembic are only generated once and intended to be slightly modified - when you add/modify existing model. This is typical case in alembic (similarly in Django's ORM) that it is "mostly generated". After generating such migration script you modify it manually to account for different dialects or some exceptional cases, data migration etc.  and only after that you commit such scripts (and possibly modify them later if you need to add some fixes in the future). 

So I think we are perfectly fine.

Reply:
In this particular case my "guess" was right: 

The "wrong" in was manually added in this commit when mssql support was added: https://github.com/apache/airflow/commit/d03cc1784b3bb0d33dfd62d451f495144f7b4af7

It was added as tuple in another file: and then limited to mssql only with tuple's comma lost along the way :)

Reply:
Got it. Thank you very much for the explanation. :)

So to me it looks ready to be 🚢 ped now.

Reply:
👍 Anyone cares for an approval :) ?